### PR TITLE
Inject stylesheets at the start of the head instead of the end. 

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: CSS scripts are now injected at the beginning of the document head, allowing for easier overriding of editor styles.

### CODE REVIEW

File: tests/utils/injectStylesheet.test.ts [hunk 1]
```diff
@@ -68,4 +73,66 @@ describe( 'injectStylesheet', () => {
 			`Stylesheet with "${ CDN_MOCK_STYLESHEET_URL }" href is already present in DOM!`
 		);
 	} );
+
+	it( 'should inject the stylesheet at the end of the head if headPlacement = \'end\'', async () => {
```
Comment (gorzelinski) [inline tests/utils/injectStylesheet.test.ts:77]: That's off-topic. I know we require single quotes. But in such cases, doesn't this seem cleaner? Or it's just me?
```suggestion
	it( "should inject the stylesheet at the end of the head if headPlacement = 'end'", async () => {
```

File: src/utils/injectStylesheet.ts [hunk 2]
```diff
@@ -30,6 +31,30 @@ export function injectStylesheet( href: string ): Promise<void> {
 		maybePrevStylesheet.remove();
 	}
 
+	// Append the link tag to the head.
+	const appendLinkTagToHead = ( link: HTMLLinkElement ) => {
+		const previouslyInsertedStylesheets = Array.from(
+			document.head.querySelectorAll( 'link[data-injected-by="ckeditor-integration"][rel="stylesheet"]' )
+		);
+
+		switch ( headPlacement ) {
```
[the remark this answers is not in the document]
Comment (gorzelinski) [reply src/utils/injectStylesheet.ts:40]: Tbh, I even wondered if somebody cares about stylesheet placement at all. But, at the same time, we have reasonable default, so it's better to have options, I guess.

[the remark this answers is not in the document]
Comment (Mati365) [reply src/utils/injectStylesheet.ts:40]: Yep, especially that it's placed in commons package. tbh I'm not sure if these options are enough - some of plugins may want to append styles in some other different positions (I hope not). At this moment I'll keep it at it is

File: src/utils/injectStylesheet.ts [hunk 3]
```diff
@@ -12,10 +12,11 @@ export const INJECTED_STYLESHEETS = new Map<string, Promise<void>>();
 /**
  * Injects a stylesheet into the document.
  *
- * @param href The URL of the stylesheet to be injected.
+ * @param props.href The URL of the stylesheet to be injected.
+ * @param props.headPlacement The placement of the stylesheet in the head.
```
[the remark this answers is not in the document]
Comment (Mati365) [reply src/utils/injectStylesheet.ts:16]: Yep, `placementInHead` sounds better. Updated.

File: tests/utils/injectStylesheet.test.ts (hunk 1 above)
Comment (gorzelinski) [reply tests/utils/injectStylesheet.test.ts:77]: I got you. I was thinking more about updating these rules. But that's off-topic.

**Review by gorzelinski — APPROVED** [review]

**Pull request merged**